packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -71,156 +71,80 @@ const defaultProps = {
   zIndex: '6000',
 };
 
-class AbstractModal extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      modalTrigger: undefined,
-    };
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.showModalDomUpdates = this.showModalDomUpdates.bind(this);
-    this.hideModalDomUpdates = this.hideModalDomUpdates.bind(this);
-    this.modalElement = React.createRef();
-  }
-
-  componentDidMount() {
+const AbstractModal = (props) => {
+  const {
+    ariaLabel,
+    children,
+    classNameModal,
+    classNameOverlay,
+    closeOnEsc,
+    closeOnOutsideClick,
+    isFullscreen,
+    isOpen,
+    role,
+    rootSelector,
+    onRequestClose,
+    zIndex,
+    ...customProps
+  } = props;
+
+  const modalElementRef = createRef();
+
+  useLayoutEffect(() => {
     // eslint-disable-next-line no-prototype-builtins
     if (!Element.prototype.hasOwnProperty('inert')) {
       // IE10 throws an error if wicg-inert is imported too early, as wicg-inert tries to set an observer on document.body which may not exist on import
       // eslint-disable-next-line global-require
       require('wicg-inert/dist/inert');
     }
-    document.addEventListener('keydown', this.handleKeydown);
-
-    if (this.props.isOpen) {
-      this.showModalDomUpdates();
-    }
-  }
-
-  componentDidUpdate(prevProps) {
-    if (this.props.isOpen) {
-      if (!prevProps.isOpen) {
-        this.showModalDomUpdates();
-      }
-    } else if (prevProps.isOpen) {
-      this.hideModalDomUpdates();
-    }
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
-    this.hideModalDomUpdates();
-  }
-
-  showModalDomUpdates() {
-    const mainDocumentElement = document.querySelector(this.props.rootSelector);
-    // Store element that was last focused prior to modal opening
-    this.setState({ modalTrigger: document.activeElement });
-
-    if (mainDocumentElement) {
-      const inert = +mainDocumentElement.getAttribute('data-overlay-count');
-
-      if (!mainDocumentElement.hasAttribute('data-overlay-count')) {
-        mainDocumentElement.setAttribute('data-overlay-count', '1');
-        mainDocumentElement.setAttribute('inert', '');
-      } else if (mainDocumentElement && mainDocumentElement.hasAttribute('data-overlay-count')) {
-        mainDocumentElement.setAttribute('data-overlay-count', `${inert + 1}`);
-      }
-
-      // Handle focus shift for VoiceOver on iOS
-      if ('ontouchstart' in window) {
-        this.modalElement.current.querySelector('[data-terra-abstract-modal-begin]').focus();
-      } else {
-        // Shift focus to modal dialog
-        this.modalElement.current.focus();
+  }, []);
+
+  useEffect(() => {
+    function handleKeydown(e) {
+      if (e.keyCode === KEY_ESCAPE && closeOnEsc && isOpen) {
```

Comment:
fun fact - according to the mozilla documentation, `keyCode` is now deprecated in favor of `code`. unfortunately `code` is not supported by edge or ie, yet `keyCode` has full browser support.  just something to keep in mind for the future

https://github.com/cerner/terra-core/issues/2866
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

Reply:
oh man.... whyyyyyy 😢